### PR TITLE
Fixes #1628 missing signature list

### LINKED ISSUE #1628: NameError: global name 'signature_list' is not defined
```python
Traceback (most recent call last):
  File "/Users/abhishesh/Dropbox/WORK/web2py/gluon/restricted.py", line 216, in restricted
    exec(ccode, environment)
  File "/Users/abhishesh/Dropbox/WORK/web2py/applications/artemis/models/db.py", line 231, in <module>
    auth.define_tables(username=False, signature=False, enable_tokens=True)
  File "/Users/abhishesh/Dropbox/WORK/web2py/gluon/tools.py", line 2238, in define_tables
    settings.table_token_name, []) + signature_list
NameError: global name 'signature_list' is not defined
```

### PR DESCRIPTION
**Pull request by leonelcamara**

### CODE REVIEW

File: gluon/tools.py [hunk 1]
```diff
@@ -2214,7 +2214,19 @@ def define_tables(self, username=None, signature=None, enable_tokens=False,
...
+        if signature is True:
+            signature_list = [self.signature]
+        elif not signature:
+            signature_list = []
+        elif isinstance(signature, Table):
+            signature_list = [signature]
+        else:
+            signature_list = signature
+
+        super(Auth, self).define_tables(username, signature_list, migrate, fake_migrate)
```
Comment (mdipierro) [inline gluon/tools.py]: why is the code above necessary? authapi.define_tables already has the same exact code

Comment (leonelcamara) [inline gluon/tools.py]: The problem is that Auth.define_tables also needs to have signature_list available if it wants to define more tables than AuthAPI. Would you prefer that AuthAPI.define_tables sets an auth instance variable with the signature list (e.g. self._table_signature_list)?

**Comment by mdipierro** [comment]
yes. that would be better. thanks

**Comment by leonelcamara** [comment]
I submitted a better version with your suggestions.

**Comment by mdipierro** [comment]
Thanks!

**Pull request closed without merging**